src/Sylius/Bundle/ShopBundle/EventListener/OrderCustomerIpListener.php

```diff
@@ -15,23 +15,34 @@
 
 use Sylius\Bundle\CoreBundle\Assigner\IpAssignerInterface;
 use Sylius\Component\Core\Model\OrderInterface;
-use Symfony\Component\EventDispatcher\GenericEvent;
 use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Workflow\Event\Event;
 use Webmozart\Assert\Assert;
 
 final class OrderCustomerIpListener
 {
-    public function __construct(private IpAssignerInterface $ipAssigner, private RequestStack $requestStack)
-    {
+    public function __construct(
+        private readonly IpAssignerInterface $ipAssigner,
+        private readonly RequestStack $requestStack
+    ) {
     }
 
-    public function assignCustomerIpToOrder(GenericEvent $event): void
+    public function __invoke(Event|OrderInterface $event): void
     {
-        $subject = $event->getSubject();
-        Assert::isInstanceOf($subject, OrderInterface::class);
+        if ($event instanceof Event) {
+            $subject = $event->getSubject();
+            Assert::isInstanceOf($subject, OrderInterface::class);
+        }
+
+        if ($event instanceof OrderInterface) {
+            $subject = $event;
+        }
```

Comment:
Since there's more logic now, we could extract order resolving to a private method, and change the property to just `$order` since `$subject` makes meagre sense now